src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -184,14 +184,11 @@ UCC_CLASS_CLEANUP_FUNC(ucc_tl_ucp_context_t)
 {
     ucc_status_t status;
     tl_info(self->super.super.lib, "finalizing tl context: %p", self);
-    while (UCC_OK != (status = ucc_tl_ucp_close_eps(self))) {
-        //TODO can we hang the runtime this way ?
-        if (status < 0) {
-            tl_error(self->super.super.lib,
-                     "failed to close ucp endpoint: %s",
-                     ucc_status_string(status));
-            break;
-        }
+    status = ucc_tl_ucp_close_eps(self);
+    if (status < 0) {
+        tl_error(self->super.super.lib,
+                 "close ucp endpoint ended with status: %s",
+                 ucc_status_string(status));
```

Comment:
it seems return status is only the last failed close ep in the list of ep.  print the status during the error in closing the specific ep at  tl_ucp_ep.c:74 log